website/docs-scraper/index-website-content.mjs

```diff
@@ -0,0 +1,328 @@
+        const records = await populateAlgoliaRecords({
+          record: currBaseRecord,
+          content: tabContent,
+        });
+        // add the page title to the record's hierarchy
+        records.forEach(
+          (record) => (record.hierarchy.lvl1 = pageMetadata.title)
+        );
+        // add the records to the algolia list
+        algoliaRecords.push(...records);
+      });
+    } else {
+      // there are no tabs, all the content is directly in the page
+      const currBaseRecord = _.merge({}, algoliaBaseRecord, {
+        searchResultURL: `/${pageURL}`, // for simplicity we want to store absolute paths
+        pageTab: null,
+      });
+      const records = await populateAlgoliaRecords({
+        record: currBaseRecord,
+        content: pageContent,
+      });
+      // add the page title to the record's hierarchy
+      records.forEach((record) => (record.hierarchy.lvl1 = pageMetadata.title));
+      // add the records to the algolia list
+      algoliaRecords.push(...records);
+    }
+  }
+
+  // --------------------------------
+  // TOKENS
+  // --------------------------------
+
+  // read the JSON file for the tokens
+  const tokensJsonData = await fs.readJSON(config.tokensJsonFilePath);
+
+  tokensJsonData.forEach((token) => {
+    algoliaRecords.push({
+      // RECORD
+      objectID: `record__token-${token.name}`, // https://www.algolia.com/doc-beta/guides/sending-and-managing-data/send-and-update-your-data#unique-object-identifiers
+      // PAGE
+      pageMainCategory: 'foundations',
+      pageSection: 'foundations',
+      // pageTitle: ???,
+      // pageCaption: ???,
+      pageBaseURL: '/foundations/tokens',
+      searchResultURL: `/foundations/tokens?searchQuery=${token.name}`,
+      // CONTENT
+      source: 'token',
+      type: 'token',
+      content: `${token.name} ${token.value}`,
+      level: 9,
+      // EXTRA
+      'token-name': token.name,
+      'token-value': token.value,
+      'token-type': token.type, // notice: it may be `undefined`
+      'token-group': token.group, // notice: it may be `undefined`
+    });
+  });
+
+  console.log(chalk.white(`\nProcessed ${tokensJsonData.length} tokens`));
+
+  // --------------------------------
+  // ICONS
+  // --------------------------------
+
+  // read the JSON file for the Flight icons
+  const flightIconsJsonData = await fs.readJSON(config.flightIconsJsonFilePath);
+
+  const distinctIcons = _.uniqBy(flightIconsJsonData.assets, 'iconName');
+
+  distinctIcons.forEach((icon) => {
+    algoliaRecords.push({
+      // RECORD
+      objectID: `record__icon-${icon.iconName}`, // https://www.algolia.com/doc-beta/guides/sending-and-managing-data/send-and-update-your-data#unique-object-identifiers
+      // PAGE
+      pageMainCategory: 'foundations',
+      pageSection: 'icons',
+      // pageTitle: ???,
+      // pageCaption: ???,
+      pageBaseURL: '/icons/library',
+      searchResultURL: `/icons/library?searchQuery=${icon.iconName}`,
```

Comment:
this is where it would be helpful if we could actually link to _only_ a given icon and not have to rely on search

Reply:
yes, I know :(